mesheryctl/pkg/utils/auth.go

```diff
@@ -342,18 +343,41 @@ func initiateRemoteProviderAuth(provider Provider) (string, error) {
 
 	// Redirect user to the provider page
 	if err := NavigateToBrowser(uri); err != nil {
+		// Terminate the temporary server if there's an error
+		if err := srv.Shutdown(context.Background()); err != nil {
+			return "", err
+		}
 		return "", err
 	}
 
-	// Pause until we get the response on the channel
-	token := <-tokenChan
-
-	// Shut down the server
-	if err := srv.Shutdown(context.TODO()); err != nil {
-		return token, err
+	// Pause until we get the response on the channel or timeout occurs
+	select {
+	case token := <-tokenChan:
+		// Shut down the server
+		if err := srv.Shutdown(context.Background()); err != nil {
+			return "", err
+		}
+		return token, nil
+	case <-time.After(5 * time.Second): // Set the timeout duration as needed
```

Comment:
@yash37158 This does not solve the issue. The issue required that `mesheryctl system login` be successful when an active meshery cloud session was already present in the browser.

Also this line causes the bug where `mesheryctl system login` finishes execution before the login is completed on the following page. This is invalid behaviour.

<img width="1399" alt="image" src="https://github.com/meshery/meshery/assets/76054921/b3cbed41-360a-498f-8324-d7d586322680">